From 042fabde578b14aff53d32e5a086debe6a17abc0 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 28 Jun 2016 17:18:17 +0800
Subject: [PATCH 1690/1691] MLK-12946 media: pxp-v4l2: correct the 32 bpp pixel
 format passed to pxp

commit 042fabde578b14aff53d32e5a086debe6a17abc0 from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

The 32 bpp pixel format which is passed to pxp should be
'PXP_PIX_FMT_RGB32' instead of 'PXP_PIX_FMT_RGB24', since
only 'PXP_PIX_FMT_RGB32' can be recognized by lcdif.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit c98efc59e1bc6d1814b2179f1b7e9f22cb177f47)
---
 drivers/media/platform/mxc/output/mxc_pxp_v4l2.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c b/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
index 17f3c07..b3bc181 100644
--- a/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
+++ b/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
@@ -106,7 +106,7 @@ static unsigned int v4l2_fmt_to_pxp_fmt(u32 v4l2_pix_fmt)
 	u32 pxp_fmt = 0;
 
 	if (v4l2_pix_fmt == V4L2_PIX_FMT_RGB24)
-		pxp_fmt = PXP_PIX_FMT_RGB24;
+		pxp_fmt = PXP_PIX_FMT_RGB32;
 	else if (v4l2_pix_fmt == V4L2_PIX_FMT_RGB565)
 		pxp_fmt = PXP_PIX_FMT_RGB565;
 	else if (v4l2_pix_fmt == V4L2_PIX_FMT_RGB555)
@@ -478,7 +478,7 @@ static int pxp_s_output(struct file *file, void *fh,
 	pxp->pxp_conf.out_param.width = fmt->width;
 	pxp->pxp_conf.out_param.height = fmt->height;
 	if (fmt->pixelformat == V4L2_PIX_FMT_RGB24)
-		pxp->pxp_conf.out_param.pixel_fmt = PXP_PIX_FMT_RGB24;
+		pxp->pxp_conf.out_param.pixel_fmt = PXP_PIX_FMT_RGB32;
 	else
 		pxp->pxp_conf.out_param.pixel_fmt = PXP_PIX_FMT_RGB565;
 
-- 
1.9.1

